src/deployment/registration.py

```diff
@@ -646,33 +646,34 @@ def set_app_audience(
         raise Exception(err_str)
 
 
-def add_user(objectId: str, principalId: str) -> None:
+def add_users(objectId: str, principalIds: List[str]) -> None:
     # Get principalId by retrieving owner for SP
-    http_body = {
-        "principalId": principalId,
-        "resourceId": objectId,
-        "appRoleId": "00000000-0000-0000-0000-000000000000",
-    }
-    try:
-        query_microsoft_graph(
-            method="POST",
-            resource="users/%s/appRoleAssignments" % principalId,
-            body=http_body,
-        )
-    except GraphQueryError:
-        query = (
-            "az rest --method post --url "
-            "https://graph.microsoft.com/v1.0/users/%s/appRoleAssignments "
-            "--body '%s' --headers \"Content-Type\"=application/json"
-            % (principalId, http_body)
-        )
-        logger.warning(
-            "execute the following query in the azure portal bash shell and "
-            "run deploy.py again : \n%s",
-            query,
-        )
-        err_str = "Unable to add user to SP using Microsoft Graph Query API. \n"
-        raise Exception(err_str)
+    for id in principalIds:
+        http_body = {
+            "principalId": id,
+            "resourceId": objectId,
+            "appRoleId": "00000000-0000-0000-0000-000000000000",
```

Comment:
is this supposed to be an approle of the onefuzz instance ?

Reply:
I'll be honest I'm not entirely sure. The information I have present is all I need to lock down the SP and add the user. If i need to specify an appRoleId for best practices, I can look into doing that.

Reply:
this is where we define [the roles](https://github.com/microsoft/onefuzz/blob/fb13fb45c1c1d888625fc5f20c9bed53831fd3e0/src/deployment/deploy.py#L295) of the application .
let's add a new role to that list and use it here.
you can look at [assign_app_role](https://github.com/microsoft/onefuzz/blob/fb13fb45c1c1d888625fc5f20c9bed53831fd3e0/src/deployment/deploy.py#L295) for an example of how to look for a specific app role and assign it